quickwit/quickwit-datetime/src/date_time_format.rs

```diff
@@ -94,12 +112,18 @@ impl StrptimeParser {
     }
 
     pub fn parse_date_time(&self, date_time_str: &str) -> Result<OffsetDateTime, String> {
-        if *self.borrow_with_timezone() {
-            OffsetDateTime::parse(date_time_str, self.borrow_items()).map_err(|err| err.to_string())
+        if let TimezoneSource::Default(default_tz) = self.borrow_timezone_source() {
+            let primitive_datetime = self
+                .parse_primitive_date_time(date_time_str)
```

Comment:
We can drop the self referencing bullshit by using the OwnedFormatItem of time rs